src/supa/connection/provider/server.py

```diff
@@ -194,13 +232,41 @@ def ReserveAbort(
         Returns:
             A response telling the caller we have received its abort request.
         """
-        log = logger.bind(method="ReserveAbort")
-        logger.info("Received message.", request_message=pb_reserve_abort_request)
+        connection_id = UUID(pb_reserve_abort_request.connection_id)
+        log = logger.bind(method="ReserveAbort", connection_id=str(connection_id))
+        log.debug("Received message.", request_message=pb_reserve_abort_request)
 
-        from supa import scheduler
+        from supa.db.session import db_session
 
-        scheduler.add_job(ReserveAbortJob(UUID(pb_reserve_abort_request.connection_id)))
+        with db_session() as session:
+            reservation = (
+                session.query(model.Reservation).filter(model.Reservation.connection_id == connection_id).one_or_none()
+            )
+            if reservation is None:
+                log.info("Connection ID does not exist")
+                send_service_exception(
+                    NsiException(
+                        ReservationNonExistent, str(connection_id), {Variable.CONNECTION_ID: str(connection_id)}
+                    )
+                )
+            else:
+                try:
+                    rsm = ReservationStateMachine(reservation, state_field="reservation_state")
+                    rsm.reserve_abort_request()
+                except TransitionNotAllowed as tna:
+                    log.info("Not scheduling ReserveAbortJob", reason=str(tna))
+                    send_service_exception(
+                        NsiException(
+                            InvalidTransition,
+                            str(tna),
+                            {Variable.RESERVATION_STATE: reservation.reservation_state},
+                        )
+                    )
+                else:
+                    from supa import scheduler
```

Comment:
Top level import

Reply:
Some components cannot be imported globally because they depend on the global settings that are not yet initialised properly.